src/Microsoft.Crank.AzureDevOpsWorker/Program.cs

```diff
@@ -160,57 +160,70 @@ private static async Task MessageHandler(ProcessMessageEventArgs args)
+                            {
+                                throw new Exception($"{LogNow} Job timed out ({jobPayload.Timeout}). The timeout can be increased in the queued message.");
+                            }
 
-                            if (!success)
+                            var logs = driverJob.FlushStandardOutput().ToArray();
+
+                            // Send any page of logs to the AzDo task log feed
+                            if (logs.Any())
                             {
-                                Console.ForegroundColor = ConsoleColor.DarkYellow;
-                                Console.WriteLine($"{LogNow} SendTaskLogFeedsAsync failed. If the task was canceled, this jobs should be stopped.");
-                                Console.ResetColor();
+                                var success = await devopsMessage.SendTaskLogFeedsAsync(String.Join("\r\n", logs));
 
-                                driverJob.Stop();
-                            }                                
-                        }
+                                if (!success)
+                                {
+                                    Console.ForegroundColor = ConsoleColor.DarkYellow;
+                                    Console.WriteLine($"{LogNow} SendTaskLogFeedsAsync failed. If the task was canceled, this jobs should be stopped.");
+                                    Console.ResetColor();
+
+                                    driverJob.Stop();
+                                }
+                            }
 
-                        // Check if task is still active (not canceled)
+                            // Check if task is still active (not canceled)
 
-                        records = await devopsMessage.GetRecordsAsync();
+                            records = await devopsMessage.GetRecordsAsync();
 
-                        // This can return a stale value (see DevopsMessage.RecordsCacheTimeSpan)
+                            // This can return a stale value (see DevopsMessage.RecordsCacheTimeSpan)
 
-                        record = records.Value.FirstOrDefault(x => x.Id == devopsMessage.TaskInstanceId);
+                            record = records.Value.FirstOrDefault(x => x.Id == devopsMessage.TaskInstanceId);
 
-                        if (record != null && record?.State == "completed")
-                        {
-                            Console.WriteLine($"{LogNow} Job is completed ({record.Result}), interrupting...");
+                            if (record != null && record?.State == "completed")
+                            {
+                                Console.WriteLine($"{LogNow} Job is completed ({record.Result}), interrupting...");
+
+                                driverJob.Stop();
+                            }
 
-                            driverJob.Stop();
+                            await Task.Delay(TaskLogFeedDelay);
```

Comment:
Should this be in an `else`? No need to delay if we stopped the driver right?

Reply:
Better